tests/integration/__init__.py

```diff
@@ -12,13 +12,15 @@ def retry(retry_num, retry_sleep_sec):
     def decorator(func):
         @functools.wraps(func)
         def wrapper(*args, **kwargs):
-            for _ in range(retry_num):
-                try:
-                    return func(*args, **kwargs)
-                except Exception:
-                    time.sleep(retry_sleep_sec)
-            logging.error("func %s retry failed", func)
-            raise Exception("Exceed max retry num: {} failed".format(retry_num))
+            try:
+                for _ in range(retry_num):
+                    try:
+                        return func(*args, **kwargs)
+                    except Exception:
+                        time.sleep(retry_sleep_sec)
+                logging.error("func %s retry failed", func)
+            except Exception as e:
+                raise Exception("Exceed max retry num: {} failed".format(retry_num)) from e
```

Comment:
This still loses the actual inner exception. I don't think you need two layers of try/except. What about this? (tested locally)

```python
def retry(retry_num, retry_sleep_sec):
    def decorator(func):
        @functools.wraps(func)
        def wrapper(*args, **kwargs):
            for i in range(retry_num):
                try:
                    return func(*args, **kwargs)
                except Exception as exc:
                    if i >= retry_num - 1:
                        raise Exception(f'Exceeded {retry_num} retries') from exc
                    logging.error("func %s failure %d/%d: %s", func.__name__, i+1, retry_num, exc)
                    time.sleep(retry_sleep_sec)

        return wrapper

    return decorator
```